### PR TITLE
fix for #2349

### PR DESCRIPTION
**Pull request by nikivazou**
The refinement type of the wrapper had an extra type variable (following the design of the worker) which made the type inconsistent with the unrefined type. This edit filters out the unused type variables to ensure the generated refinement types is good.

### CODE REVIEW

File: tests/pos/T2349.hs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{-# LANGUAGE GADTs #-}
+{-@ LIQUID "--reflection" @-}
+module TestModule where
```
Comment (facundominguez) [inline tests/pos/T2349.hs]: The test needs to be listed in `tests/tests.cabal`, or it won't run in CI.

Comment (facundominguez) [reply tests/pos/T2349.hs]: Also, it is better to name the module the same as the file.

Comment (facundominguez) [inline tests/pos/T2349.hs]:
```suggestion
-- | Test that the refinement types produced for GADTs are
-- compatible with the Haskell types.
module T2349 where
```

File: tests/pos/T2349.hs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  NIL :: Expr [t]
+  CONS :: Int -> Expr t -> Expr [t] -> Expr [t]
+
+{-@
+data Expr t where
+  I :: Int -> Expr Int
+  NIL :: Expr {v:[t] | len v = 0}
+  CONS :: xxxxx:_ -> head:Expr t -> tail:Expr {v:[t] | len v = xxxxx} ->
+          Expr {v:[t] | len v = xxxxx+1}
```
Comment (facundominguez) [inline tests/pos/T2349.hs]: Should the test include a use of CONS? e.g.

```Haskell
{-@
fail e
e :: Expr {v:[Int] | len v = 0}
@-}
e :: Expr [Int]
e = CONS 0 (I 2) NIL
```

Comment (nikivazou) [reply tests/pos/T2349.hs]: Edited! (I added a negative test instead)

**Comment by facundominguez** [comment]
Thanks Niki. I hooked the tests in `tests.cabal`.

What does reflection have to do with the example? I could check that disabling reflection makes the tests fail, but I don't know why.

**Comment by nikivazou** [comment]
I am not sure why reflection is needed to be honest... But I believe it does not affect this PR :)

**Pull request merged**